src/yunohost/tools.py

```diff
@@ -162,6 +162,45 @@ def tools_maindomain(auth, new_domain=None):
         logger.warning("%s" % e, exc_info=1)
         raise MoulinetteError(errno.EPERM, m18n.n('maindomain_change_failed'))
 
+    # Set hostname
+    
+    commands = [     
+        "sudo hostnamectl --static set-hostname %s" % new_domain,
+        "sudo hostnamectl --transient set-hostname %s" % new_domain,
+        "sudo hostnamectl --pretty set-hostname (YunoHost/%s)" % new_domain
```

Comment:
Reviving this discussion : not sure what to do here. The current code works, and I don't see any benefits of using directly a list (I already do a .split() when using subprocess right after - the only limitation is that we cannot have spaces in the pretty hostname atm, but we don't really care imho)